### ISSUE TITLE
Accessibility issues while testing datagrid using AXE accessibility checker[ https://www.deque.com/axe/ ]

### ISSUE THREAD

**Issue by pavandixit93** [vmware/clarity#4736]
## Describe the bug
We are using clarity datagrid in our project. As per requirement we are using `AXE accessibility checker` https://www.deque.com/axe/, a chrome extension for accessibility testing. It gives the ARIA and accessibility related issue if present. 
When we ran AXE extension on datagrid example, we found 2 critical severity issues. Please find attached snippets.

Issue found in,
1. **column toggle picker button in clr-dg-column-toggle component** 
-> As per observation, here `aria-owns` and `aria-controls` is used but `aria-expanded` is not used as column-picker is opened and closed on button click.
![column-toggle--action-btn_ARIA_InValid_Value](https://user-images.githubusercontent.com/21095558/84896407-8a148a00-b0c1-11ea-9b21-80ebdfcf4bc5.png)

2. **SVG icons** used in  clr-dg-pagination component buttons.
-> As per observation, here `role=img` is used on SVG icons but no `title` or `aria-label` or 'aria-labelledby` is used.
![pagination-btn-SVG_ARIA_no_accessible_text](https://user-images.githubusercontent.com/21095558/84896426-94368880-b0c1-11ea-8303-36f2e1faca58.png)

_A clear and concise description of what the bug is, and the conditions when it occurs._

## How to reproduce

_Please provide a link to a reproduction scenario using one of the [Clarity Stackblitz templates](https://stackblitz.com/@clr-team). Reports without a clear reproduction may not be prioritized until one is provided._

Steps to reproduce the behavior `SVG icons present in clr-dg-pagination component`:
1.  Install `axe - Web Accessibility Testing` chrome extension.(https://www.deque.com/axe/)
2.  Go to 'https://clarity.design/documentation/datagrid/pagination'
3.  Open browser console(inspect mode)
4.  Click on `axe` tab
5.  Click on `ANALYZE` button.

Steps to reproduce the behavior `column toggle picker button in clr-dg-column-toggle component`:
1.  Install `axe - Web Accessibility Testing` chrome extension.
2.  Go to 'https://clarity.design/documentation/datagrid/hide-show'
3.  Open browser console(inspect mode)
4.  Click on `axe` tab
5.  Click on `ANALYZE` button.

## Expected behavior

_A clear and concise description of what you expected to happen._

## Versions

**App**

- Angular: [e.g. 6]
- Node: [e.g. 8.10.0]
- Clarity: [e.g. 0.12.5]

**Device:**

- Type: [e.g. MacBook]

## Additional notes

_Add any other notes about the problem here._

**Comment by gnomeontherun** [comment]
Hello, are these just automated testing concerns or have you also done manual testing? If so, please provide more context about the expectations and results.

**Comment by pavandixit93** [comment]
Hi @gnomeontherun ,
Error comes for both automated and manual testing.
**Expectation:**
1. column toggle picker button in clr-dg-column-toggle component -> **aria-expanded is missing.**
-> As per observation, here aria-owns and aria-controls is used but aria-expanded is not used as column-picker is opened and closed on button click. 
Just for your information, Error is not coming in case of filter icon button, there you have used aria-expanded along with aria-controls and aria-owns.

2. SVG icons used in clr-dg-pagination component buttons. -> **aria-label OR aria-labelledby OR title is missing**
-> As per observation, here role=img is used on SVG icons but no title or aria-label or 'aria-labelledby` is used. aria-label is added on button element and SVG is inside the button element.

**Comment by pavandixit93** [comment]
@gnomeontherun Hi,
Any update on this please? Let me know if you need any details further.
I have already shared my findings.

**Comment by pavandixit93** [comment]
@gnomeontherun Hi,
Any update on this please? Let me know if you need any details further.
I have already shared my findings.

**Comment by gnomeontherun** [comment]
Hi @pavandixit93 are you asking on behalf of VMware? The accessibility team has to audit all issues and align them to our standards, so we haven't confirmed that this is really an issue yet.

**Comment by scroniser** [comment]
@gnomeontherun & @pavandixit93 - it looks like this issue is being addressed through [another issue](https://github.com/vmware/clarity/issues/4454). If I'm mistaken please let me know and I'll investigate further.

**Comment by pavandixit93** [comment]
@scroniser Hi, Yes i have mentioned two use cases. the second use case is related to SVG missing title issue which can be addressed through https://github.com/vmware/clarity/issues/4454 .

**Comment by pavandixit93** [comment]
@gnomeontherun No i am not asking behalf of VMware. I am using clarity in my project. 
Let me be more technical here,
column toggle picker button in clr-dg-column-toggle component
-> As per observation, here aria-owns and aria-controls is used but aria-expanded is not used as column-picker is opened and closed on button click.
Here as per `a11y` standard, aria-expanded is missing, which should be added and updated base on popper toggling state.
This is added in clr-dg-string-filter component.

**Comment by scroniser** [comment]
@gnomeontherun - I can see the issue and confirm that it's problematic. This is element incorrectly uses `aria-owns` on the button element. This essentially renders the menu items invisible to assistive technology. As explained in this [GitHub post by a11y expert & W3C ARIA chair Matt King](https://github.com/w3c/html-aria/issues/124#issuecomment-451726237), placing `aria-owns` on the button element essentially renders the unordered list elements as it's children.

@pavandixit93 is correct, this element needs the `aria-expanded attribute`. The toggle column filter uses a correct `aria-expanded` implementation.

FWIW, @gnomeontherun, I saw this issue when following-up to a question posted in one of our internal a11y Slack channels. This was not part of an internal audit.

**Comment by pavandixit93** [comment]
@scroniser Thank you for confirmation,I hope, this one and SVG will be fixed soon.

**Comment by mathisscott** [comment]
@pavandixit93 
Thanks for the continued interest in this issue. It is one we will be addressing but our priorities are focused on delivering 4.0 right now. If our near term objectives do not align with your product team's wants, you could submit a contribution. Updating aria attributes seems like a good candidate for helping out if you need this effort resolved quickly.

If/when you do submit a PR, though, I'd recommend pinging @chlane so that our a11y lead has a chance to vet the approach/effort. Thanks!

**Comment by chlane** [comment]
@gnomeontherun and @scroniser I agree with Santina's proposed fix of adding and dynamically updating aria-expanded as necessary.

**Comment by gnomeontherun** [comment]
Clarity cares deeply about accessibility, and we work with our accessibility team to identify and triage issues. We have an internal tracking system to identify and verify accessibility issues, which has captured this already. Some issues with Clarity Angular are difficult to solve without causing side effects or breaking changes, and we have to balance concerns with fixing issues and breaking changes. As we build out Clarity Core, we are able to address these root cause accessibility issues and recommend adopting the Clarity Core implementations as they become available. We are closing this issue as we have the issues tracked internally. Please follow our development and releases to see when we release relevant Clarity Core components to make this possible.
